connectors/sources/servicenow.py

```diff
@@ -0,0 +1,346 @@
+            )
+        )
+
+    async def close_session(self):
+        """Closes unclosed client session"""
+        self._sleeps.cancel()
+        if self.session is None:
+            return
+        await self.session.close()
+        self.session = None
+
+
+class ServiceNowDataSource(BaseDataSource):
+    """ServiceNow"""
+
+    name = "ServiceNow"
+    service_type = "servicenow"
+
+    def __init__(self, configuration):
+        """Setup the connection to the ServiceNow instance.
+
+        Args:
+            configuration (DataSourceConfiguration): Instance of DataSourceConfiguration class.
+        """
+
+        super().__init__(configuration=configuration)
+        self.concurrent_downloads = self.configuration["concurrent_downloads"]
+        self.servicenow_client = ServiceNowClient(configuration=configuration)
+
+    def tweak_bulk_options(self, options):
+        """Tweak bulk options as per concurrent downloads support by ServiceNow
+
+        Args:
+            options (dict): Config bulker options.
+        """
+
+        options["concurrent_downloads"] = self.concurrent_downloads
+
+    @classmethod
+    def get_default_configuration(cls):
+        """Get the default configuration for ServiceNow.
+
+        Returns:
+            dict: Default configuration.
+        """
+        return {
+            "url": {
+                "label": "Service URL",
+                "order": 1,
+                "type": "str",
+                "value": "http://127.0.0.1:9318",
+            },
+            "username": {
+                "label": "Username",
+                "order": 2,
+                "type": "str",
+                "value": "admin",
+            },
+            "password": {
+                "label": "Password",
+                "sensitive": True,
+                "order": 3,
+                "type": "str",
+                "value": "changeme",
+            },
+            "services": {
+                "label": "List of services",
+                "order": 4,
+                "type": "list",
+                "value": "*",
+            },
+            "retry_count": {
+                "display_value": RETRIES,
+                "display": "numeric",
+                "label": "Maximum retries per request",
+                "order": 5,
+                "required": False,
+                "type": "int",
+                "ui_restrictions": ["advanced"],
+                "value": RETRIES,
+            },
+            "concurrent_downloads": {
+                "default_value": MAX_CONCURRENCY_SUPPORT,
+                "display": "numeric",
+                "label": "Maximum concurrent downloads",
+                "order": 6,
+                "required": False,
+                "type": "int",
+                "ui_restrictions": ["advanced"],
+                "value": MAX_CONCURRENCY_SUPPORT,
+            },
+        }
+
+    async def _remote_validation(self):
```

Comment:
Should this be decorated with `@retryable`?

Reply:
Retry mechanism is already handle internally in `_api_call`